bqskit/qis/unitary/unitarymatrix.py

```diff
@@ -119,7 +119,8 @@ def get_distance_from(self, other: UnitaryLike) -> float:
         other = UnitaryMatrix(other)
         num = np.abs(np.trace(other.get_numpy().conj().T @ self.get_numpy()))
         dem = self.get_dim()
-        dist = np.sqrt(1 - ((num / dem) ** 2))
+        val = 1 - ((num / dem) ** 2)
+        dist = np.sign(val) * np.sqrt(np.abs(val))
```

Comment:
`val` should not be absoluted. If the value is negative, we should round up to zero.

`dist = np.sqrt(val)`
The return statement on the next line will round up to zero.

Reply:
Changed this back. I switched it to this in the first place because numpy was giving warnings about square rooting negatives.

Reply:
Yeah that makes sense, we could round up to zero before the sqrt; gives the same effect but no warnings.